SapphireDb/SapphireDbContext.cs

```diff
@@ -12,9 +12,9 @@ namespace SapphireDb
 {
     public class SapphireDbContext : DbContext
     {
-        private readonly SapphireDatabaseNotifier notifier;
+        private readonly ISapphireDatabaseNotifier notifier;
 
-        public SapphireDbContext(DbContextOptions options, SapphireDatabaseNotifier notifier) : base(options)
+        public SapphireDbContext(DbContextOptions options, ISapphireDatabaseNotifier notifier) : base(options)
```

Comment:
This will probably cause an Exception from the ASP.NET Core's DI framework because in 
`SapphireDatabaseExtensions` we are not registering this dependency as `ISapphireDatabaseNotifier`.

I think in `SapphireDatabaseExtensions` we need to add the line
```c#
services.AddScoped<ISapphireDatabaseNotifier, SapphireDatabaseNotifier>();
```
after
```c#
services.AddScoped<SapphireDatabaseNotifier>();
```

The old line is still necessary for compatibility reasons: there are probably many apps with own db context implementations asking for `SapphireDatabaseNotifier` which would break without that line.

You should also change the `README.md` and `README-NUGET.md` to reflect the new best-practice.

`AuthDemoContext` and `DemoContext` may be updated accordingly too.

Reply:
Resolved by PR author via an additional commit.